R/extremes-data.R

```diff
@@ -1,169 +1,190 @@
+      
+    })
+    
+    names(min.max) <- paste0("data$", names(data)[i], "$", names(subset))
+    results <- append(results, min.max) 
+    
+  }
+  
+  
+  maximums <- results[grep("max", names(results))]
+  maximums_index <- c(grep("upchain", names(maximums)), 
+                      grep("primary", names(maximums)), 
+                      grep("dv", names(maximums)))
+  maximums <- maximums[maximums_index]
+  
+  minimums <- results[grep("min", names(results))]
+  minimums_index <- c(grep("upchain", names(minimums)), 
+                      grep("primary", names(minimums)), 
+                      grep("dv", names(minimums)))
+  minimums <- minimums[minimums_index]
+  
+  results <- list()
+  results <- append(results, maximums)
+  results <- append(results, minimums)
+  
+  #Change column and row names to their correct forms and add them into the dataframe.
+  toRet <- data.frame()
+  for(i in 1:length(results)){
+    toAdd <- cbind(c(orderedRowNames[i],rep("",nrow(results[[i]])-1)),results[[i]]) 
+    colnames(toAdd) <- columnNames
+    rownames(toAdd) <- NULL
+
+    toRet <- rbind(toRet,toAdd)
+  }
+  
+  return(toRet)
+}
+
+applyQualifiers <- function(data) {
+  consolidatedQualifiers <- list(
+    primary=data$primary$qualifiers, 
+    upchain=data$upchain$qualifiers,
+    dv=data$dv$qualifiers)
+  
+  return(sapply(data, function(x) {
+    if(! is.null(x$qualifiers)) {
+      x$max$points <- applyQualifiersToValues(x$max$points, x$qualifiers)
+      x$min$points <- applyQualifiersToValues(x$min$points, x$qualifiers)
+      x$max$relatedUpchain <- applyQualifiersToValues(x$max$relatedUpchain, consolidatedQualifiers$upchain)
+      x$min$relatedUpchain <- applyQualifiersToValues(x$min$relatedUpchain, consolidatedQualifiers$upchain)
+      x$max$relatedPrimary <- applyQualifiersToValues(x$max$relatedPrimary, consolidatedQualifiers$primary)
+      x$min$relatedPrimary <- applyQualifiersToValues(x$min$relatedPrimary, consolidatedQualifiers$primary)
+    }
+    return(x)
+  }))
+}
+
+applyQualifiersToValues <- function(points, qualifiers) {
+  if(is.null(points)) return(points)
+  
+  getQualifierString <- function(p) {
+    builtQualifiers <- ""
+    if(length(qualifiers) > 0) {
+      for(i in 1:nrow(qualifiers)) {
+        q <- qualifiers[i,]
+        startDate <- q$startDate
+        endDate <- q$endDate
+        if(p$time > startDate & p$time < endDate) {
+          builtQualifiers <- paste0(builtQualifiers, q$code, ",")
+        }
+      }
+      strLength <- nchar(builtQualifiers)
+      if(strLength > 0) {
+        builtQualifiers <- substr(builtQualifiers, 1, strLength-1)
+      }
+    }
+    return(builtQualifiers)
+  }
+  
+  points <- mutate(points, 
+         value = paste(getQualifierString(points), points$value))
+  return(points)
+}
+
+flattenParam <- function(param){
+  baseParam <- strsplit(gsub("([A-Z])", " \\1", param[1]), " ")[[1]]
+  param <- paste(unique(c(baseParam, param[-1])), collapse='')
+  return(param)
+}

```

Comment:
Did you change anything in these files? Changing line endings has made any differences hard to see.